src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/MLDsa/MLDsaTestHelpers.cs

```diff
@@ -0,0 +1,448 @@
+                    DoTryUntilDone((Span<byte> destination, out int bytesWritten) =>
+                        mldsa.TryExportEncryptedPkcs8PrivateKey(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(password)), pbeParameters, destination, out bytesWritten)));
+
+                callback((mldsa, password, pbeParameters) =>
+                    mldsa.ExportEncryptedPkcs8PrivateKey(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(password)), pbeParameters));
+
+                callback((mldsa, password, pbeParameters) =>
+                    DecodePem(mldsa.ExportEncryptedPkcs8PrivateKeyPem(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(password)), pbeParameters)));
+            }
+
+            static byte[] DecodePem(string pem)
+            {
+                PemFields fields = PemEncoding.Find(pem.AsSpan());
+                Assert.Equal(Index.FromStart(0), fields.Location.Start);
+                Assert.Equal(Index.FromStart(pem.Length), fields.Location.End);
+                Assert.Equal("ENCRYPTED PRIVATE KEY", pem.AsSpan()[fields.Label].ToString());
+                return Convert.FromBase64String(pem.AsSpan()[fields.Base64Data].ToString());
+            }
+        }
+
+        internal delegate string ExportToPemCallback(MLDsa mldsa, string password, PbeParameters pbeParameters);
+        internal static void AssertExportToEncryptedPem(
+            Action<ExportToPemCallback> callback,
+            EncryptionPasswordType passwordTypesToTest = EncryptionPasswordType.All)
+        {
+            if ((passwordTypesToTest & EncryptionPasswordType.Char) != 0)
+            {
+                callback((mldsa, password, pbeParameters) =>
+                    mldsa.ExportEncryptedPkcs8PrivateKeyPem(password, pbeParameters));
+                callback((mldsa, password, pbeParameters) =>
+                    mldsa.ExportEncryptedPkcs8PrivateKeyPem(password.AsSpan(), pbeParameters));
+            }
+
+            if ((passwordTypesToTest & EncryptionPasswordType.Byte) != 0)
+            {
+                callback((mldsa, password, pbeParameters) =>
+                    mldsa.ExportEncryptedPkcs8PrivateKeyPem(new ReadOnlySpan<byte>(Encoding.UTF8.GetBytes(password)), pbeParameters));
+            }
+        }
+
+        internal static void AssertExportToPrivateKeyPem(Action<Func<MLDsa, string>> callback) =>
+            callback(mldsa => mldsa.ExportPkcs8PrivateKeyPem());
+
+        internal static void AssertExportToPublicKeyPem(Action<Func<MLDsa, string>> callback) =>
+            callback(mldsa => mldsa.ExportSubjectPublicKeyInfoPem());
+
+        internal delegate bool TryExportFunc(Span<byte> destination, out int bytesWritten);
+        internal static byte[] DoTryUntilDone(TryExportFunc func)
+        {
+            byte[] buffer = new byte[512];
+            int written;
+
+            while (!func(buffer, out written))
+            {
+                Array.Resize(ref buffer, buffer.Length * 2); 
```

Comment:
Nit: "View changes since last review" says a random space got added at the end of this line.

```suggestion
                Array.Resize(ref buffer, buffer.Length * 2);
```